anomaly_detectors/anomalib_lmi/anomaly_model2.py

```diff
@@ -141,53 +140,138 @@ def preprocess(self, image):
+            user_inference_batch_size > 0
+        )
+        
+        if perform_mini_batch_inference:
+            all_mini_batch_outputs = []
+            for i in range(0, num_samples_in_input, user_inference_batch_size):
+                mini_batch = input_batch[i:min(i + user_inference_batch_size, num_samples_in_input)]
+                current_mini_batch_output_tensor = None
+                if self.inference_mode == 'TRT':
+                    self.binding_addrs['input'] = int(mini_batch.data_ptr())
+                    self.context.execute_v2(list(self.binding_addrs.values()))
+                    current_mini_batch_output_tensor = self.bindings['output'].data
+                               
+                elif self.inference_mode == 'PT':
+                    preds = self.pt_model(mini_batch)
+                    if isinstance(preds, torch.Tensor):
+                        current_mini_batch_output_tensor = preds
+                    elif isinstance(preds, dict):
+                        current_mini_batch_output_tensor = preds['anomaly_map']
+                    elif isinstance(preds, Sequence):
+                        current_mini_batch_output_tensor = preds[1]
+                    else:
+                        raise Exception(f'Unknown prediction type from PT model: {type(preds)}')
+                
+                if current_mini_batch_output_tensor is not None:
+                    all_mini_batch_outputs.append(current_mini_batch_output_tensor)
+                else:
+                    raise Exception(f"Model failed to produce an output for a mini-batch.")
+
+            if not all_mini_batch_outputs:
+                raise Exception("Batched inference was performed, but no outputs were collected.")
+
+            if isinstance(all_mini_batch_outputs[0], torch.Tensor):
+                aggregated_output_tensor = torch.cat(all_mini_batch_outputs, dim=0)
             else:
-                raise Exception(f'Unknown prediction type: {type(preds)}')
-            
+                raise Exception(f"Unsupported output type for aggregation: {type(all_mini_batch_outputs[0])}")
+
+        else: 
+            if self.inference_mode == 'TRT':
+                self.binding_addrs['input'] = int(input_batch.data_ptr())
+                self.context.execute_v2(list(self.binding_addrs.values()))
+                aggregated_output_tensor = self.bindings['output'].data
+
+            elif self.inference_mode == 'PT':
+                preds = self.pt_model(input_batch)
+                if isinstance(preds, torch.Tensor):
+                    aggregated_output_tensor = preds
+                elif isinstance(preds, dict):
+                    aggregated_output_tensor = preds['anomaly_map']
+                elif isinstance(preds, Sequence):
+                    aggregated_output_tensor = preds[1]
+                else:
+                    raise Exception(f'Unknown prediction type: {type(preds)}')
+        
```

Comment:
It seems that two chunks of code (lines 239-253 and 209-223)  are repetitive. They can be simplified into a function and call that function twice.